Index: gstreamer1.0/libs/gst/helpers/ptp/ffi.rs
===================================================================
--- gstreamer1.0.orig/libs/gst/helpers/ptp/ffi.rs
+++ gstreamer1.0/libs/gst/helpers/ptp/ffi.rs
@@ -698,6 +698,39 @@ pub mod unix {
         ))]
         pub type time_t = c_long;
 
+        #[cfg(all(
+            target_os = "linux",
+            target_env = "gnu",
+            not(any(
+                target_arch = "x86",
+                target_arch = "x86_64",
+                target_arch = "arm",
+                target_arch = "aarch64",
+                target_arch = "m68k",
+                target_arch = "mips",
+                target_arch = "mips64",
+                target_arch = "powerpc",
+                target_arch = "powerpc64",
+                target_arch = "sparc",
+                target_arch = "sparc64",
+                target_arch = "s390x",
+                target_arch = "riscv32",
+                target_arch = "riscv64",
+                target_arch = "loongarch64",
+            )),
+        ))]
+        pub type time_t = c_long;
+
+        #[cfg(all(
+            target_os = "linux",
+            not(any(
+                target_env = "gnu",
+                target_env = "musl",
+                target_env = "uclibc",
+            )),
+        ))]
+        pub type time_t = c_long;
+
         #[repr(C)]
         pub struct timespec {
             pub tv_sec: time_t,
